docs/source/architecture/events_and_transactions_receipts.rst

```diff
@@ -312,13 +312,13 @@ the TransactionReceiptStore after a block is validated.
 
 Transaction receipts will only be stored in this off-chain store and will not
 be included in the block. Note that because a transaction may exist in multiple
-blocks at a time, the transaction receipt is stored by both transaction id and
+blocks at a time, the transaction receipt is stored with both transaction id and
```

Comment:
The TransactionReceipt is updated upon a block commit. We may not have a scenario that at same height multiple blocks are committed. So this becomes an invalid scenario. We could have same transaction in multiple blocks but they are not stored in TransactionReceiptStore until a block is committed. That means when transaction is stored in the TransactionReceipt it only can be referred by TxnID.

put() & get() methods do not require state_root_hash. This also resonates with the latest codebase.